File: Day1/day1.hs
{-# LANGUAGE OverloadedStrings #-}

import Data.String
import Data.Char
import Data.Text(pack, unpack, replace)

main :: IO ()
main = do
  contents <- readFile "input"
  putStrLn (show (sum (processLines(lines contents))))

replaceNumbers :: String -> String
replaceNumbers = unpack . replace "one" "o1ne" .
                          replace "two" "t2o".
                          replace "three" "t3e".
                          replace "four" "f4r".
                          replace "five" "f5e".
                          replace "six" "s6x".
                          replace "seven" "s7n".
                          replace "eight" "e8t".
                          replace "nine" "n9n". pack

processLines :: [String] -> [Int]
processLines [] = []
processLines (x:xs) = (processLine(replaceNumbers x)) : (processLines xs)

processLine :: String -> Int
processLine x = 
    let nums = numbers(x) in
        let beginning = head(nums) in 
            let end = last(nums) in 
    read ([beginning] ++ [end]) :: Int

numbers :: String -> [Char]
numbers "" = []
numbers (l:ls) = if ord l >= 48 && ord l <= 57
    then (l : (numbers ls))
    else (numbers ls)

File: Day01/day1.py
f = open("Day01/input", "r")
numbers = []
for x in f:
  s = x.strip()
  s = s.replace("one", "o1ne") \
            .replace("two", "t2wo") \
            .replace("three", "t3hree") \
            .replace("four", "f4our") \
            .replace("five", "f5ive") \
            .replace("six", "s6ix") \
            .replace("seven", "s7even") \
            .replace("eight", "eigh8t") \
            .replace("nine", "nin9e") 
  nums = []
  for c in s:
    if c.isnumeric():
      nums.append(c)
  if len(nums) == 1:
    numbers.append(int(nums[0] + nums[0]))
  else:
    numbers.append(int(nums[0] + nums[-1]))

print(sum(numbers))
